RoyaltiesProcessorDelivery/src/index.ts

```diff
@@ -128,9 +135,20 @@ async function processSales(
   const exchangeFilter = salesFilter.exchangeFilter
 
   let sales: T_Sale[]
-  if (!useOpenSeaApi) {
+  if (!useOpenSeaApi && !useReservoirApi) {
     // get all sales of all tokens in blockrange in subgraph from subgraph
     sales = await saleService.getAllSalesBetweenBlockNumbers(blockRange)
+  } else if (useReservoirApi) {
+    if (
+      salesFilter.contractFilterType !== 'ONLY' ||
+      salesFilter.contractsFilter === undefined
+    ) {
+      throw 'ONLY filter for subset of contracts is required when using OpenSea API mode'
+    }
```

Comment:
Is this applicable when using the reservoir api?

Reply:
Currently, I actually think it is - no logic in there yet to filter out contracts. Nothing on reservoir's side, just some internal logic stuff

This should be tackled in the "PBAB" TODO